Mirror pages to github

struan opened the pull request:
This runs over all the pages on the site, saving them to a local cache directory and then using everypoliticianbot's `with_git_repo` magic commits them to the mirror-data branch of this repository.

Files are saved as a SHA of the absolute URL with the session data stripped out - see `Mirror.sha_url` for details of this. If a file with the same SHA as the URL exists it will not save the page. It will still visit the page though.

The actual scraping works by visiting every person page listed on the all people page of the site (http://www.congreso.es/portal/page/portal/Congreso/Congreso/Diputados/DiputadosTodasLegislaturas) and from there visiting the page that lists all the terms that person was elected to and visiting all the person pages linked from there.

Once it has done a complete sweep of all the pages it commits them, or if there is an exception then it commits what it has.

I've left in all the old code that scraped and parsed the existing site, although it might be easier to delete most of it, but at least the CSS/XPath bits might be useful.

Improvements that could be made:
- [ ] Commit as we go so that each file is commited once it's been saved. I've not done this as the current way `everypoliticianbot` works doesn't allow this - if you move the `with_git_repo` wrapper round the file writing then it falls over
- [ ] Better checks that the pages it's getting are the pages we expect. Sometime the site responds with error pages but they still have a 200 status code.
- [ ] Rescraping existing files. If a file with the same SHA as the URL exists it currently moves on to the next page. This means we won't pick up changes at the moment
- [ ] Maybe a mode where it only visits a page if it doesn't exist in the cache. I'm not sure how well this would work with Capybara.

struan committed:
Mirror pages to a cache directory

struan committed:
commit scraped files to github reposistory

Also, move Capybara include inside mirror class otherwise it and
everypoliticianbot disagree and you end up with a stack too deep error

tmtmtmtm commented on scraper.rb:
```diff
@@ -33,6 +36,67 @@ def tidy
+    File.open(path,"w") do |f|
+      f.write(page.html)
+    end
+    sleep(1)
```
Why do we need to sleep after writing?

struan replied:
Er, because the site has seemed quite fragile in the past so I was being cautious about sending lots of requests. And because it's polite :) This may be old fashioned of me.

tmtmtmtm replied:
Isn't that taken care off with the 

``` ruby
    # the website is a bit fragile to lets not hammer it with requests
    sleep(2)
```

below?

Sleeping in the loop seems fine (where anyone can choose their own approach), but forcing a sleep inside the `save_page` method doesn't seem right.

struan committed:
save all the pages

save the intermediate pages as well as the people pages.

Move the visit call into the save_page method and use that to visit all
the pages so we save them as we visit them.
Rename save_page to vist_and_save_page becuase that's what it does.
Also simplify the URL SHAing and push it out to a function to it can be
used separately from the saving.

struan committed:
only save pages that return a 200 status

struan committed:
only parse term page if we find a link for it

struan committed:
catch errors with scraping and commit what we have so far

As the scraper is prone to falling over due to issues with the site it
is scraping wrap the mirror process in a begin/rescue and spit out a
message if it falls over. This means that it should also commit the
progress it's made to github rather than it being an all or nothing
approach.
